api/dependencies/auth.py

```
@@ -0,0 +1,142 @@
+        # Verify the token
+        decoded_token = auth.verify_id_token(token)
+        logger.info(f"Token verified for user: {decoded_token.get('uid')}")
+        return decoded_token
+    except Exception as e:
+        logger.error(f"Invalid token: {str(e)}")
+        raise HTTPException(
+            status_code=401,
+            detail=f"Invalid authentication credentials: {str(e)}"
+        )
+
+async def get_current_user(token: dict = Depends(verify_token)) -> dict:
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)